aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1610,25 +1668,29 @@ impl VMValidator for AptosVM {
 
         let executor_view = state_view.as_executor_view();
         let resolver = self.as_move_resolver(&executor_view);
-        let mut session = self.0.new_session(&resolver, SessionId::prologue(&txn));
-        let validation_result = self.validate_signature_checked_transaction(
+        let txn_data = TransactionMetadata::new(&txn);
+
+        let mut session = self.new_session(&resolver, SessionId::prologue_meta(&txn_data));
+        let validation_result = self.validate_signed_transaction(
             &mut session,
             &resolver,
             &txn,
-            true,
```

Comment:
why were we passing true here and now false?

Reply:
Before: there was a field `allow_too_new` to deal with sequence numbers, which was true for both validation and simulation. I moved the check out to validation function. But itself the function is calling `run_prologue` which can be different for simulation (multisigs) and execution.